x/coin.go

```diff
@@ -51,13 +51,62 @@ func (c Coin) ID() string {
 	return c.Issuer + "/" + c.Ticker
 }
 
+// Split divides the value of a coin into given amount of pieces and returns a
+// single piece.
+// It might be that a precise splitting is not possible. Any leftover of a
+// fractional value is returned as well.
+// For example splitting 4 EUR into 3 pieces will result in a single piece
+// being 1.33 EUR and 1 cent returned as the rest (leftover).
+//   4 = 1.33 x 3 + 1
+func (c Coin) Divide(pieces int64) (one, rest Coin) {
+	// This is an invalid use of the method.
+	if pieces <= 0 {
+		zero := Coin{Ticker: c.Ticker}
+		return zero, zero
+	}
+
+	// When dividing whole and there is a leftover then convert it to
+	// fractional and split as well.
+	fractional := c.Fractional
+	if leftover := c.Whole % pieces; leftover != 0 {
+		fractional += leftover * FracUnit
+	}
+
+	one = Coin{
+		Ticker:     c.Ticker,
+		Whole:      c.Whole / pieces,
+		Fractional: fractional / pieces,
+	}
+	rest = Coin{
+		Ticker:     c.Ticker,
+		Whole:      0, // This we can always divide.
+		Fractional: fractional % pieces,
+	}
+	return one, rest
+}
```

Comment:
req: There is a bug in this code. I have looked for the root cause but dividing factional 2 by 3 returns the wrong result. See my test in https://github.com/iov-one/weave/pull/322

Reply:
Done in https://github.com/iov-one/weave/pull/316/commits/97945acd3d2169852c9898d771d60c43e9b461cd

I believe the original test was wrong. Dividing 20 cents between two parties with weight 1 and 2 should give 6 and 12 accodringly, leaving a leftover of 2. 
Orignal test was expecting the above to return 6 and 13.